### PR TITLE
Feature gloo federation

### PR DESCRIPTION
**Pull request by ashleywang1**
# Description

This is a collection of glooctl and Failover changes necessary to make Gloo work with Gloo Federation.

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: install/helm/gloo/templates/19-gloo-mtls-configmap.yaml [hunk 1]
```diff
@@ -16,14 +16,43 @@ data:
...
         - filters:
           - name: envoy.tcp_proxy
             config:
               stat_prefix: ingress_tcp
-              cluster: gloo_cluster
+              cluster: gloo_xds_cluster
+          transport_socket:
+            name: envoy.transport_sockets.tls
+            typed_config:
+              "@type": type.googleapis.com/envoy.api.v2.auth.DownstreamTlsContext
```
Comment (kdorosh) [inline install/helm/gloo/templates/19-gloo-mtls-configmap.yaml:30]: let's use v3 if we can

Comment (kdorosh) [reply install/helm/gloo/templates/19-gloo-mtls-configmap.yaml:30]: `type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext`

File: install/helm/gloo/templates/19-gloo-mtls-configmap.yaml [hunk 2]
```diff
@@ -16,14 +16,43 @@ data:
...
+                      api_type: GRPC
+                      grpc_services:
+                      - envoy_grpc:
+                          cluster_name: gloo_sds
+      - name: envoy_rest_xds_mtls_listener
+        address: { socket_address: { address: "0.0.0.0", port_value: {{ .Values.gloo.deployment.restXdsPort }} } }
+        filter_chains:
+        - filters:
+          - name: envoy.tcp_proxy
+            config:
```
Comment (kdorosh) [inline install/helm/gloo/templates/19-gloo-mtls-configmap.yaml:53]: same, lets use v3
e.g.
```
- name: envoy.filters.network.tcp_proxy
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
```

File: install/helm/gloo/templates/9-gateway-proxy-configmap.yaml [hunk 3]
```diff
@@ -179,6 +179,45 @@ data:
...
+        upstream_connection_options:
+          tcp_keepalive: {}
+        type: STRICT_DNS
+        respect_dns_ttl: true
+{{- if $global.glooMtls.enabled }}
+        transport_socket:
+          name: envoy.transport_sockets.tls
+          typed_config:
+            "@type": type.googleapis.com/envoy.api.v2.auth.UpstreamTlsContext
+            common_tls_context:
```
Comment (kdorosh) [inline install/helm/gloo/templates/9-gateway-proxy-configmap.yaml:204]: update to v3

**Comment by arianaw66** [comment]
Could we leave stuff as v2 and just let this PR do the update: https://github.com/solo-io/gloo/pull/3343

File: install/helm/gloo/templates/9-gateway-proxy-configmap.yaml (hunk 3 above)
Comment (kdorosh) [reply install/helm/gloo/templates/9-gateway-proxy-configmap.yaml:204]: `"@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext`

File: pkg/version/enterprise_version_test.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ entries:
 		Expect(err).To(BeNil())
 		_, err = tmpFile.WriteString(fileString)
 		Expect(err).To(BeNil())
-		enterpriseVersion, err := version.LatestVersionFromRepo(tmpFile.Name(), true)
+		enterpriseVersion, err := version.LatestVersionFromRepo(tmpFile.Name(), "gloo-ee", true)
```
Comment (kdorosh) [inline pkg/version/enterprise_version_test.go:43]: is "gloo-ee" a constant already somewhere in the code? if not we should make it one

File: projects/gloo/cli/pkg/cmd/install/installer.go [hunk 5]
```diff
@@ -123,9 +137,11 @@ func (i *installer) Install(installerConfig *InstallerConfig) error {
 		return err
 	}
 
-	// We need this to avoid rendering the CRDs we include in the /templates directory
-	// for backwards-compatibility with Helm 2.
-	setCrdCreateToFalse(installerConfig)
+	if !(installerConfig.Mode == Federation) {
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/cmd/install/installer.go:140]: nit, more idiomatic to have
```golang
	if installerConfig.Mode != Federation {
```

Comment (kdorosh) [reply projects/gloo/cli/pkg/cmd/install/installer.go:140]: Is Gloo Fed supporting helm2?

File: projects/gloo/cli/pkg/cmd/install/installer_test.go [hunk 6]
```diff
@@ -147,19 +148,25 @@ rules:
...
-			Version:         "test",
-			CreateNamespace: true,
+			HelmInstall: options.HelmInstall{
+				Namespace:       defaults.GlooSystem,
+				HelmReleaseName: constants.GlooReleaseName,
+				Version:         "test",
+				CreateNamespace: true,
+			},
+		}
+		if federation || mode == install.Federation {
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/cmd/install/installer_test.go:160]: can we rename `federation` to `isFederated` or something that more clearly implies it's a bool value

Comment (ashleywang1) [reply projects/gloo/cli/pkg/cmd/install/installer_test.go:160]: These bool values actually aren't needed.

**Pull request closed without merging**